app/controllers/orders_controller.rb

```diff
@@ -33,8 +34,30 @@ def index
                           .per(Settings.length.digit_8)
   end
 
+  def cancel
+    if @order.pending?
+      @order.cancel!
+      flash[:success] = t "order.cancel_success"
+    else
+      flash[:danger] = t "order.cancel_failed"
+    end
+  rescue ActiveRecord::RecordInvalid
+    flash[:danger] = t "order.cancel_failed"
+    redirect_back(fallback_location: root_path)
+  else
+    redirect_back(fallback_location: root_path)
+  end
```

Comment:
có sự lặp chỗ này, sao lại ko dùng ensure nhỉ 

```suggestion
  ensure
    redirect_back(fallback_location: root_path)
  end
```